daemon/image_delete.go

```diff
@@ -105,26 +105,34 @@ func (daemon *Daemon) ImageDelete(imageRef string, force, prune bool) ([]types.I
 		repoRefs = daemon.referenceStore.References(imgID)
 
 		// If this is a tag reference and all the remaining references
-		// to this image are digest references, delete the remaining
-		// references so that they don't prevent removal of the image.
+		// to the same repository are digest references, delete the
+		// remaining repository references so that they don't prevent
+		// removal of the image.
```

Comment:
Let's clarify the "so they don't prevent removal of the image" part.

As I understand it, there may well be other references to the image, so we aren't necessarily deleting it. In many cases, we'll be removing the digest for cleanliness.